# This file is part of HiGFXback

# requires
REQUIRES="lite-build meson-build mupdf-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists djvulibre-build && REQUIRES="$REQUIRES djvulibre-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists poppler-build && REQUIRES="$REQUIRES poppler-build"

# configure
PKG_CONFIG_PATH=/dfb/lib/pkgconfig meson setup --prefix=/dfb build

# build
meson compile -C build

# install
install -d $DESTDIR/dfb/bin
install build/src/projektor $DESTDIR/dfb/bin
install -d $DESTDIR/dfb/share/projektor
install -m 644 data/progress_off.dfiff $DESTDIR/dfb/share/projektor
install -m 644 data/progress_on.dfiff $DESTDIR/dfb/share/projektor

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/projektor-build.pc << EOF
Name: Projektor
Version: 20250405
Description: DirectFB document viewer
Requires: $REQUIRES

exec=\\
/dfb/bin/projektor \\
/dfb/share/projektor/progress_off.dfiff \\
/dfb/share/projektor/progress_on.dfiff
EOF

exit

# patch
--- Projektor-20250405.orig/README
+++ Projektor-20250405/README
@@ -1,4 +1,4 @@
 Projektor
 =========
 
-Projektor is a document viewer runnning on DirectFB.
+Projektor is a document viewer running on DirectFB.

# source
https://github.com/directfb2/Projektor/archive/a5cc7b8e0eb746b4a7294f7eaf6d2e85733352e8/Projektor-20250405.tar.gz
